packages/core/src/contrib/Minkowski.ts

```diff
@@ -138,13 +140,44 @@ export const convexPolygonMinkowskiSDF = (
 };
 
 /**
- * Returns list of convex polygons comprising the original polygon.
- * @param p Sequence of points defining a polygon.
+ * Returns list of convex polygons comprising the original polygon. Assumes that
+ * the polygon shape remains fixed after this function is called; that is, some
+ * transformations can be applied, but vertices cannot change independently.
+ * @param p Sequence of points defining a simple polygon.
  */
 export const convexPartitions = (p: VarAD[][]): VarAD[][][] => {
-  // TODO: Add convex partitioning algorithm for polygons.
-  // See e.g.: https://doc.cgal.org/Manual/3.2/doc_html/cgal_manual/Partition_2/Chapter_main.html#Section_9.3
-  return [p];
+  if (p.length < 3) {
+    return [p];
+  }
+
+  // map each point back to its original VecAD object; note, this depends on the
+  // fact that two arrays with the same contents are considered different as
+  // keys in a JavaScript Map, very scary!
+  const pointMap = new Map(p.map((point) => [point.map(numOf), point]));
+
+  const polygon = [...pointMap.keys()];
+  // https://www.npmjs.com/package/poly-decomp/v/0.3.0#advanced-usage
+  if (!decomp.isSimple(polygon)) {
+    throw Error("convex partitioning requires polygon to be simple");
+  }
+  decomp.makeCCW(polygon);
+  const convexPolygons: number[][][] = decomp.quickDecomp(polygon);
+
+  const covered: Set<number[]> = new Set(); // again using object identity 😱
```

Comment:
Comments with emojis! 😱 Technology has finally gone too far.. 😄

Reply:
Haha should I remove it? 😛 (e.g. in case @keenancrane's preferred editor can't render emoji)

Reply:
No, let's embrace the future :D